src/formatDate.js

```diff
@@ -9,6 +9,46 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const fromDelimeter = fromFormat[fromFormat.length - 1];
+  const dateArr = date.split(fromDelimeter);
+  const dataObj = {};
+
+  for (let d = 0; d <= 2; d++) {
+    dataObj[fromFormat[d]] = dateArr[d];
+  }
+
+  const properObj = {};
+  const propFormat = toFormat.slice(0, -1);
+
+  for (const d of propFormat) {
+    if (d === 'YY' || d === 'YYYY') {
+      properObj[d] = dataObj['YY'] || dataObj['YYYY'];
+    } else {
+      properObj[d] = dataObj[d];
+    }
+  }
+
+  if (('' + properObj['YY']).length === 4) {
+    properObj['YY'] = ('' + properObj['YY']).slice(2);
+  }
```

Comment:
This condition checks if the 'YY' property is a four-digit year, but it will never be true because the 'YY' property is only set if the year is two digits long. This code is redundant and should be removed.